klasses/utils.py

```diff
@@ -0,0 +1,152 @@
+"""Utility functions for Klasses"""
+
+from klasses.models import BootcampRunCertificate, BootcampRunEnrollment
+
+
+def generate_single_certificate(user, bootcamp_run):
+    """Generates certificate for a single user if its enrollment is active"""
+    result = {"updated": False}
+    if user and bootcamp_run:
+        certificate = BootcampRunCertificate.all_objects.filter(
+            user=user, bootcamp_run=bootcamp_run
+        ).first()
+        if certificate:
+            result[
+                "msg"
+            ] = "A {} certificate:{} already exists for user:{} in bootcamp-run:{}".format(
+                "revoked" if certificate.is_revoked else "",
+                certificate.link,
+                user.email,
+                bootcamp_run,
+            )
+        elif BootcampRunEnrollment.objects.filter(
+            user=user, bootcamp_run=bootcamp_run, active=True
+        ).exists():
+            certificate = BootcampRunCertificate.objects.create(
```

Comment:
will it not create multiple entries for same `user` and `bootcamp_run`. Not better to use `get_or_create/update_or_create ` there

Reply:
On line 10, we have already checked if a certificate already exists or not for this user and bootcamp-run.